extensions/kinesis/src/main/java/com/hazelcast/jet/kinesis/KinesisSources.java

```diff
@@ -286,16 +290,40 @@ public Builder withInitialShardIteratorRule(@Nonnull String shardIdRegExp,
          */
         @Nonnull
         public StreamSource<Map.Entry<String, byte[]>> build() {
+            FunctionEx<? super Record, byte[]> projectionFn = Builder::toArray;
+            return build(projectionFn);
+        }
+
+        /**
+         * Constructs the source based on the options provided so far and with user-specified projection function.
+         * @param projectionFn projection function that will convert Amazon's {@link Record} into some user-defined type
+         *                     as the map value.
+         * @since 5.1
+         */
+        @Nonnull
+        public <T> StreamSource<Map.Entry<String, T>> build(FunctionEx<? super Record, ? extends T> projectionFn) {
```

Comment:
It's not a separate function like withProjectionFn just to maintain backward compatibility

Reply:
Do you mean that you would need to add a type parameter to  `Builder`? I think that would be fine, because the Builder constructor is private ,we only expose it via `com.hazelcast.jet.kinesis.KinesisSources#kinesis` and that could default to `Builder<byte[]>`

Reply:
Ok, so basically `withProjectionFn` would return builder with changed type parameter? It makes sense then, I'll change it